### PR TITLE
Task/2544 fix hover docstring now line available

### LINKED ISSUE #2544: Docstring for HoverTool needs updating as does now work with line
Hovertool docs say:

>Warning
>Point hit testing is not currently available on all glyphs. Hover tool currently does not work with line or image type glyphs.

I believe that line hover does now work, so this needs updating.

### PR DESCRIPTION
**Pull request by birdsarah**
issues: fixes #2544

### CODE REVIEW

File: bokeh/models/tools.py [hunk 1]
```diff
@@ -371,8 +371,22 @@ class HoverTool(Tool):
...
+            * arc
+            * bezier
+            * gear
+            * image
+            * image_rgba
+            * image_url
+            * multi_line
+            * oval
+            * patch
+            * quad
```
Comment (bryevdv) [inline bokeh/models/tools.py]: `quad` should support? 

https://github.com/bokeh/bokeh/blob/master/bokehjs/src/coffee/renderer/glyph/quad.coffee#L32

File: bokeh/models/tools.py [hunk 2]
```diff
@@ -371,8 +371,22 @@ class HoverTool(Tool):
...
+            * quadratic
+            * ray
+            * segment
+            * text
```
Comment (bryevdv) [inline bokeh/models/tools.py]: Might consider `hlist` for this (would have to build docs and see how it looks)

http://sphinx-doc.org/markup/para.html

File: bokeh/models/tools.py (hunk 1 above)
Comment (birdsarah) [inline bokeh/models/tools.py]: Doesn't work for me (you can run the updated glyphs example in this PR to see). However, having had a look it seems to break for a different reason. The others all have a console message "Point testing...." this one says: `Error: Permission denied to access property "type"` - maybe that's a regression.

Comment (bryevdv) [inline bokeh/models/tools.py]: I don't see that error. The problem is we don't actually check/enforce that `top > bottom` and `right > left`. If I make this change:
```
diff --git a/examples/glyphs/glyphs.py b/examples/glyphs/glyphs.py
index 571645f..6c24b5f 100644
--- a/examples/glyphs/glyphs.py
+++ b/examples/glyphs/glyphs.py
@@ -52,7 +52,7 @@ glyphs = [
     ("oval", Oval(x="x", y="y", width=screen(15), height=screen(25), angle=-0.7, fill_color="#1D91C0")),
     ("patch", Patch(x="x", y="y", fill_color="#A6CEE3")),
     ("patches", Patches(xs="xs", ys="ys", fill_color="#FB9A99")),
-    ("quad", Quad(left="x", right="xm01", top="y", bottom="ym01", fill_color="#B3DE69")),
+    ("quad", Quad(left="x", right="xp01", top="y", bottom="ym01", fill_color="#B3DE69")),
     ("quadratic", Quadratic(x0="x", y0="y", x1="xp02", y1="y", cx="xp01", cy="yp01", line_color="#4DAF4A", line_width=3)),
     ("ray", Ray(x="x", y="y", length=45, angle=-0.7, line_color="#FB8072", line_width=2)),
     ("rect", Rect(x="x", y="y", width=screen(10), height=screen(20), angle=-0.7, fill_color="#CAB2D6")),
``` 
the hover tool works. I would suggest making that change in this PR. 

For consistency, probably ought to do this somewhere soon too:
```
diff --git a/bokehjs/src/coffee/renderer/glyph/quad.coffee b/bokehjs/src/coffee/renderer/glyph/quad.coffee
index 6eb0f64..fcced82 100644
--- a/bokehjs/src/coffee/renderer/glyph/quad.coffee
+++ b/bokehjs/src/coffee/renderer/glyph/quad.coffee
@@ -36,8 +36,8 @@ class QuadView extends Glyph.View
 
     hits = []
     for i in [0...@sleft.length]
-      if (sx >= @sleft[i] and sx <= @sright[i] and sy >= @stop[i] and
-          sy < @sbottom[i])
+      console.log @sleft[i], sx, @sright[i], @stop[i], sy, @sbottom[i]
+      if (sx >= @sleft[i] and sx <= @sright[i] and sy >= @stop[i] and sy <= @sbottom[i])
```

Comment (bryevdv) [inline bokeh/models/tools.py]: Actually, this is a much better hit test entirely for `Quad`:
```
diff --git a/bokehjs/src/coffee/renderer/glyph/quad.coffee b/bokehjs/src/coffee/renderer/glyph/quad.coffee
index 6eb0f64..e53a035 100644
--- a/bokehjs/src/coffee/renderer/glyph/quad.coffee
+++ b/bokehjs/src/coffee/renderer/glyph/quad.coffee
@@ -31,14 +31,10 @@ class QuadView extends Glyph.View
 
   _hit_point: (geometry) ->
     [vx, vy] = [geometry.vx, geometry.vy]
-    sx = @renderer.plot_view.canvas.vx_to_sx(vx)
-    sy = @renderer.plot_view.canvas.vy_to_sy(vy)
-
-    hits = []
-    for i in [0...@sleft.length]
-      if (sx >= @sleft[i] and sx <= @sright[i] and sy >= @stop[i] and
-          sy < @sbottom[i])
-        hits.push(i)
+    x = @renderer.xmapper.map_from_target(vx, true)
+    y = @renderer.ymapper.map_from_target(vy, true)
+
+    hits = (x[4].i for x in @index.search([x, y, x, y]))
 
     result = hittest.create_hit_test_result()
     result['1d'].indices = hits
```

Comment (bryevdv) [inline bokeh/models/tools.py]: We should figure out what to do (if anything) if the user gives inconsistent data to quad (i.e., if `left > right`) we could warn, or error, or try to fix it up automatically.... not sure what is best as a policy.

Comment (bryevdv) [inline bokeh/models/tools.py]: I opened a PR to fix up the point hit testing code better. I say fix up the example file and remove quad from the list of unsupported glyphs (but continue the discussion of how to make it leas easy to mess up in the original issue #964)

**Comment by birdsarah** [comment]
Thanks for the detailed review @bryevdv. This should be ready now.

**Comment by birdsarah** [comment]
Getting a travis fail that appears to be unrelated: 

````sh
> node install.js
-Writing location.js file
PhantomJS is already installed at /usr/local/phantomjs/bin/phantomjs.
npm ERR! cb() never called!
npm ERR! not ok code 0
Command failed: /bin/bash -x -e /home/travis/build/bokeh/bokeh/conda.recipe/build.sh
The command "scripts/travis_install" failed and exited with 1 during .
````

**Pull request merged**